### PR TITLE
[python3] add vcpkg_get_vcpkg_installed_python

### PR DESCRIPTION
**Pull request by Neumann-A**
alternative to #38297

### CODE REVIEW

**Review by MonicaLiu0311 — APPROVED** [review]

File: ports/python3/vcpkg-port-config.cmake [hunk 1]
```diff
@@ -10,3 +10,53 @@ else()
 endif()
 set(PYTHON3_SITE "${site_base}/site-packages")
 
+function(get_working_installed_python out_python)
```
Comment (dg0yt) [inline ports/python3/vcpkg-port-config.cmake:13]: Does this need a `vcpkg_` prefix?

**Review by MonicaLiu0311 — DISMISSED** [review]

**Comment by vicroms** [comment]
@BillyONeal
@data-queue
@JavierMatosD
@ras0219-msft
@vicroms

We discussed this PR during our meeting and have the following comments to make:

* We think this is a good alternative to #38297
* This resolves Billy's security concerns because it doesn't add extra DLL loading for all people deploying Python bits. 
* What do you think about turning this into a separate helper port, for example, `python3-interp` with a `"supports": "native"` clause. That could also allow users to inject a system Python more easily.

File: ports/python3/vcpkg-port-config.cmake (hunk 1 above)
Comment (vicroms) [reply ports/python3/vcpkg-port-config.cmake:13]: Yeah, this should be `vcpkg_` prefixed. @BillyONeal also thinks that the `working` part is a misnomer.

File: ports/vcpkg-get-python/vcpkg-port-config.cmake [hunk 2]
```diff
@@ -0,0 +1,50 @@
+function(vcpkg_get_vcpkg_installed_python out_python)
+  if(NOT VCPKG_TARGET_IS_WINDOWS)
+    # vcpkg installed python on !windows works as normal python would work.
+    set(${out_python} "${CURRENT_HOST_INSTALLED_DIR}/tools/python3/python3" PARENT_SCOPE)
+    return()
+  endif()
+
+  if(PORT STREQUAL "python3") # Just for manual testing
```
Comment (vicroms) [inline ports/vcpkg-get-python/vcpkg-port-config.cmake:8]: Should this be removed before merging?

Comment (Neumann-A) [reply ports/vcpkg-get-python/vcpkg-port-config.cmake]: moved the required logic for testing into the python3 port.

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the new feature1

**Pull request merged**